app/buck2_execute/src/re/uploader.rs

```diff
@@ -350,9 +357,13 @@ impl Uploader {
 
         // Upload
         let upload_res = if !upload_files.is_empty() || !upload_blobs.is_empty() {
+            let mut mtd = use_case.metadata();
+            if let Some(identity) = identity {
+                apply_identity(identity, &mut mtd);
+            }
```

Comment:
Hm, so I'm not the expert on this code, but I'm not super clear on the difference between the cases where this is applied vs the cases where it isn't? Is there a reason that some of these are more important than others?

I would've expected the `Option<&ReActionIdentity>` to instead become a mandatory parameter to `use_case.metadata()`, so that call sites have to explicit ask for the `None` behavior instead of letting this go forgotten

Reply:
Great question! 
This PR covers the RE GRPC request sequence: `find missing blobs` -> `upload missing blobs` -> `execute action` -> `download produced artifacts`. 
This way on the server side we can be smart when replicating CAS artifacts and dispatching actions (we will be able to correlate`action execution` with preceeding `artifact uploads`). 
Without this change we are simply receiving 100s of parallel `find missing blobs` requests, 100s of parallel `upload missing blobs` and `execute action` without any correlation between them (even though they are correlated).
Other cases are not covered (can be covered in the future) because a) are not relevant regarding replication or request dispatching, b) will unnecessarily increase the grpc requests payload (in the header section).

Reply:
> Other cases are not covered (can be covered in the future) because a) are not relevant regarding replication or request dispatching

Leaving things not completely finished in this PR is fine, but just for clarity: What are the other cases? I guess I see action cache fetches and maybe `cas_artifact` actions. Anything I'm missing?

> will unnecessarily increase the grpc requests payload (in the header section).

I think even (and maybe especially) if this is intentionally not being filled out in some cases we should make that explicit in the code by requiring the parameter on `use_case.metadata()` and then writing a comment to explain why we pass `None` in some cases. Otherwise this will be forgotten about in the future

Reply:
That makes sense. I’ve updated the code to address this concern.